Library/Homebrew/attestation.rb

```diff
@@ -45,13 +45,13 @@ class GhAuthNeeded < RuntimeError; end
     # @api private
     sig { returns(T::Boolean) }
     def self.enabled?
-      # TODO: allow this undocumented variable until this is rolled out more
-      #       widely and then we can remove or document it.
-      return false if ENV.fetch("HOMEBREW_NO_VERIFY_ATTESTATIONS", false)
+      return false if Homebrew::EnvConfig.no_verify_attestations?
       return true if Homebrew::EnvConfig.verify_attestations?
+      return false if GitHub::API.credentials.blank?
       return false if ENV.fetch("CI", false)
+      return false unless Formula["gh"].any_version_installed?
 
-      Homebrew::EnvConfig.developer?
+      Homebrew::EnvConfig.developer? || Homebrew::EnvConfig.devcmdrun?
```

Comment:
This would result in an error if a user has an older version of `gh` installed (< `2.49.0` iirc). Perhaps we could do something like:
```
      return false unless Formula["gh"].any_version_installed?

      if Homebrew::EnvConfig.developer? || Homebrew::EnvConfig.devcmdrun?
        result = system_command!(gh_executable, args: ["--help"], print_stderr: false)
        return true if result.stdout.include?("attestation")
      end

      false
```

Or else we could force an upgrade of `gh`?

Reply:
@nandahkrishna Good idea but don't think it's worth blocking this PR over as this is verifying it's at least installed by Homebrew and various other messages will shout about this.